libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -666,84 +682,100 @@ bool PbftManager::stateOperations_() {
   duration_ = now_ - round_clock_initial_datetime_;
   elapsed_time_in_round_ms_ = std::chrono::duration_cast<std::chrono::milliseconds>(duration_).count();
 
-  auto round = getPbftRound();
-  LOG(log_tr_) << "PBFT current round is " << round << ", step is " << step_;
-
-  // Periodically verify unverified votes
-  vote_mgr_->verifyVotes(round, [this](auto const &v) {
-    const auto pk = key_manager_->get(v->getVoterAddr());
-    if (!pk) {
-      LOG(log_wr_) << "No vrf key mapped for vote author " << v->getVoterAddr();
-      return false;
-    }
-    try {
-      v->validate(dposEligibleVoteCount_(v->getVoterAddr()), getDposTotalVotesCount(), getThreshold(v->getType()), *pk);
-    } catch (const std::logic_error &e) {
-      LOG(log_wr_) << e.what();
-      return false;
+  auto [round, previous_round_period] = getPbftRoundAndPeriod();
+  LOG(log_dg_) << "PBFT current round(r): " << round << ", r-1 period: " << previous_round_period << ", step " << step_;
+
+  static bool is_new_round = false;
```

Comment:
why static?

Reply:
because I need it to hold the value throughout multiple calls. I could either have class member or static variable - as this variable is needed only in this single function I used static rather than class member

Reply:
ok, but I do not think it is good idea, as now you have some state, that it is shared between all instances of pbft manager and it should be bind to specific instance not to all. Doesn't it influence tests?

Reply:
true, fixed